jenkins/shared-library/vars/publish.groovy

```diff
@@ -72,22 +76,27 @@ void pythonPublisher(Map config, String phase) {
 		credentialsId = TEST_PYTHON_CREDENTIALS_ID
 	}
 
-	Object requirementsFile = readFile 'requirements.txt'
-	requirementsFile.readLines().each { line ->
-		runScript("poetry add ${line}")
-	}
-	runScript('cat pyproject.toml')
-
-	runScript('poetry build')
-
-	withCredentials([string(credentialsId: credentialsId, variable: 'POETRY_PYPI_TOKEN_PYPI')]) {
-		if (isAlphaRelease(phase)) {
-			runScript('poetry config "repositories.test" "https://test.pypi.org/legacy/"')
-			runScript("poetry config 'pypi-token.test' ${POETRY_PYPI_TOKEN_PYPI}")
-			runScript('poetry publish --repository "test"')
-		} else {
-			runScript("poetry config 'pypi-token.pypi' ${POETRY_PYPI_TOKEN_PYPI}")
-			runScript('poetry publish')
+	withCredentials([string(credentialsId: credentialsId, variable: 'PYPI_TOKEN')]) {
+		env.TWINE_USERNAME = '__token__'
+		env.TWINE_PASSWORD = env.PYPI_TOKEN
```

Comment:
if possible, it would be better to set these in publish.sh to avoid the twine knowledge here

if not possible, ok, but might add comment